csi/controller_test.go

```diff
@@ -755,3 +766,142 @@ func TestControllerValidateVolumeAccessModeUnknown(t *testing.T) {
 	_, err := c.ValidateVolumeCapabilities(context.Background(), req)
 	assert.NotNil(t, err)
 }
+
+func TestControllerListVolumesInvalidArguments(t *testing.T) {
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+	c := csi.NewControllerClient(s.Conn())
+
+	// Setup request
+	req := &csi.ListVolumesRequest{}
+
+	// Expect error without version
+	_, err := c.ListVolumes(context.Background(), req)
+	assert.NotNil(t, err)
+	serverError, ok := status.FromError(err)
+	assert.True(t, ok)
+	assert.Equal(t, serverError.Code(), codes.InvalidArgument)
+	assert.Contains(t, serverError.Message(), "Version")
+
+	// Expect error with maxentries set
+	// To be removed once CSI Spec issue #138 is resolved
+	req.Version = &csi.Version{}
+	req.MaxEntries = 1
+	_, err = c.ListVolumes(context.Background(), req)
+	assert.NotNil(t, err)
+	serverError, ok = status.FromError(err)
+	assert.True(t, ok)
+	assert.Equal(t, serverError.Code(), codes.Unimplemented)
+	assert.Contains(t, serverError.Message(), "token")
+}
+
+func TestControllerListVolumesEnumerateError(t *testing.T) {
+	// Create server and client connection
+	s := newTestServer(t)
+	defer s.Stop()
+	c := csi.NewControllerClient(s.Conn())
+
+	// Setup mock
+	s.MockDriver().
+		EXPECT().
+		Enumerate(gomock.Any(), gomock.Any()).
+		Return(nil, fmt.Errorf("TEST")).
+		Times(1)
+
+	// Setup request
+	req := &csi.ListVolumesRequest{
+		Version: &csi.Version{},
+	}
+
+	// Expect error without version
```

Comment:
remove/change this comment

Reply:
Interesting, I changed this line, but it doesn't show up here as changed.  See: 

https://github.com/libopenstorage/openstorage/pull/212/files#diff-a4745ba624b732fc84e4999918f9e5fbR817